beluga/include/beluga/motion/differential_drive_model.hpp

```diff
@@ -64,15 +68,15 @@ struct DifferentialDriveModelParam {
 
 /// Sampled odometry model for a differential drive.
 /**
- * This class implements OdometryMotionModelInterface2d and satisfies \ref MotionModelPage.
+ * This class satisfies \ref MotionModelPage.
  *
  * See Probabilistic Robotics \cite thrun2005probabilistic Chapter 5.4.2.
  */
 class DifferentialDriveModel {
  public:
-  /// Update type of the motion model, same as the state_type in the odometry model.
-  using update_type = Sophus::SE2d;
-  /// State type of a particle.
+  /// Current and previous odometry estimates as motion model control action.
+  using control_action_type = std::tuple<Sophus::SE2d, Sophus::SE2d>;
```

Comment:
> I was expecting [previous, current] instead of [current, previous], but this makes sense with the rolling window usage. I wonder if others will have the same expectations.

So, I think in discrete difference equations, like $y[n] = b_0 x[n] + b_1 x[n - 1] + ...$. That's why $u = (x[n], x[n - 1])$ makes sense to me, but I really don't mind. We can drop layout reversal for rolling windows and get it in reverse.

> nit^n: What do you think about control_type instead of control_action_type? Just putting it out there, I'm fine with the current choice.

I was toying with `control_type` at the beginning too. At the end, I switch to `control_action_type` because it seemed more accurate to me. But again, I don't mind strongly about it.

Reply:
> We can drop layout reversal for rolling windows and get it in reverse.

I don't think that's necessary, and it makes sense to me that we call push_front in rolling windows.

> But again, I don't mind strongly about it.

The only thing I'm thinking is that `control_type` looks nice and consistent with `state_type` IMO. But I'll leave it up to you.

Reply:
Switched to `control_type`.